.circleci/test-api.sh

```diff
@@ -0,0 +1,35 @@
+#! /usr/bin/env bash
+
+apt-get update
+apt-get install -y jq
+
+make serve-local &
+
+curl --retry-connrefused \
+    --retry 5 \
+    --retry-delay 20 \
+    http://127.0.0.1:5000 --fail
+
+var=$?
+
+if [[ $var -ne 0 ]]
+then
+  echo "Server failed to start."
+  exit $var
+fi
+
+response=$(curl --write-out '%{http_code}' --silent --output /dev/null http://127.0.0.1:5000/spec)
+if [[ $response -ne 200 ]]
+then
+  echo "Accessing the swagger endpoint returns an HTTP $response code instead of a 200 code."
+  exit 1
+fi
```

Comment:
I understand this is already what the `--fail` option does, as can be seen with `curl https://httpstat.us/400 --fail` 🙂

Reply:
From what I understand, the `--fail` option fails silently (no output at all) [on server errors](https://curl.se/docs/manpage.html), hence the following [run](https://app.circleci.com/pipelines/github/openfisca/country-template/107/workflows/5f4922e6-f97f-4314-a59b-d203f12a9dba/jobs/1092), where I purposely gave the [wrong endpoint](https://github.com/openfisca/country-template/commit/71b2c3a0ad1d825bceecc6613b2c721d160b52af#diff-74fe81df0a9ccb18afb1539db65408e43bb8924e6dbd235ddf00073e6a4200f2R9) and the test still succeeded in the CI even after having a ```curl: (22) The requested URL returned error: 404 NOT FOUND```.
[--fail-with-body ](https://curl.se/docs/manpage.html) seemed like an option to solve the problem, however it's only available for `curl 7. 76+ and I have `curl 7.64.1`.
I then [assigned the curl command to a variable](https://github.com/openfisca/country-template/commit/26712d9e371c5824a94ab244a1789f218c52f936) and that seems to solve the issue. I now get an [error](https://app.circleci.com/pipelines/github/openfisca/country-template/108/workflows/4201355b-4fbb-4b1d-b7f0-1b3b6bd1c27b/jobs/1095).
What do you think ? :)

Reply:
I don't understand why the variable assignment would be enough to change the exit code 🤔

> fails silently (no output at all)

To _fail_ means having an exit code > 0, which is different from “silently”, with no output.
In this case, it is not even _silent_ as it outputs the HTTP status code and status name. The `--fail` option only stops the server page from being shown, and changes the exit code, which is what we want.

In the current case, I guess `jq` is happy with an empty input and will not fail. We thus have to either `set -o pipefail` or just avoid the pipe with an intermediate variable:

```bash
set -e

response=$(curl ……)
jq $response
```

[More information](https://unix.stackexchange.com/a/73180).